python/cuspatial/cuspatial/core/binpreds/feature_covers.py

```diff
@@ -36,21 +49,58 @@ class CoversPredicateBase(EqualsPredicateBase):
     pass
 
 
+class LineStringLineStringCovers(BinPred):
+    def _preprocess(self, lhs, rhs):
+        return _basic_equals_all(rhs, lhs)
```

Comment:
Wait, `covers` only requires that no point of B lies out of A. It's not only the vertices coincides.
```
In [22]: l0 = LineString([(0, 0), (1, 1)])
In [23]: l1 = LineString([(0.25, 0.25), (0.5, 0.5)])
In [24]: l0.covers(l1)
Out[24]: True
```

Reply:
Thanks for pointing out this oversight, none of my tests represented this case. I've updated the predicates to reflect the new test, which made `Touches` particularly messy. :)